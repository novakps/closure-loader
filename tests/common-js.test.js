const path = require('path');
const execBundle = require('./helpers/execBundle');

const config = {
    module: {
        rules: [
            {
                test: /legacy-code\/.*\.js$/,
                use: [
                    {
                        loader: require.resolve('..'),
                        options: {
                            paths: [
                                path.join(
                                    __dirname,
                                    'fixtures',
                                    'common-js',
                                    'legacy-code',
                                ),
                            ],
                            es6mode: false,
                            watch: false,
                        },
                    },
                ],
            },
        ],
    },
};

test('loads goog modules with require()', async () => {
    const result = await execBundle('common-js', config);
    expect(result.code).toBe(0);
    expect(result.stdout).toMatchSnapshot();
});
