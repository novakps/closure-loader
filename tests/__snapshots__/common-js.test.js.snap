// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads goog modules with require() 1`] = `
"this is some.legacy.namespace.myFn
some.legacy.namespace.otherModule.text = This is from otherModule.
Some var: 42"
`;
